### DISCUSSION TITLE
useStore after await causes Cross-Request State Pollution

### DISCUSSION THREAD

**Discussion by erikmode** [vuejs/pinia#2077] [Help and Questions] [answered]
Hi! 

Im building an app with SSR using Vue3 and Pinia. I noticed a bug where state between users/requests leaked between each other if request was sent at the same time. 

It was caused by writing useStore after an await in one of my pinia action that was running on the server side, like this:
```
   //another-store.js
    async fetchUserData() {
      const someData = await fetch('https://mycoolapi.com/somedata')
      const userStore = useUserStore()
      this.data = { ...someData, user: userStore .data }
    }
```
But i later fixed the issue by moving the useUserStore above the await line like this:
```
   //another-store.js
    async fetchUserData() {
      const userStore = useUserStore()
      const someData= await fetch('https://mycoolapi.com/somedata')
      this.data = { ...someData, user: userStore.data}
    }
```

My questions is:
1. Why the state pollution happens when writing useStore after await?
2. Is there some way to avoid this mistake from happening by for example writing an ESlint rule to warn developers?

**Comment by erikmode** [comment, marked as answer]
Apperently it is mentioned here in the documentation:
https://pinia.vuejs.org/cookbook/composing-stores.html#shared-actions 

_"Since actions can be asynchronous, make sure all of your useStore() calls appear before any await. Otherwise, this could lead to using the wrong pinia instance in SSR apps"_

**Reply by rafaellages-hotmart** [reply]
An ESlint rule will be really useful as this is super easy to happen

**Reply by posva** [reply]
Agree it would be useful! It should be added to #2638
